vicon/udp.h

```diff
@@ -184,6 +187,27 @@ class UDPClient
     //----------------------------------------------------------------------------
     // Public API
     //----------------------------------------------------------------------------
+    unsigned int getNumObjects()
+    {
+        std::lock_guard<std::mutex> guard(m_ObjectDataMutex);
+        return m_ObjectData.size();
+    }
+
+    ObjectDataType getObjectData(unsigned int id)
+    {
+        std::lock_guard<std::mutex> guard(m_ObjectDataMutex);
+        if(id < m_ObjectData.size()) {
+            return m_ObjectData[id];
+        }
+        else {
+            throw std::runtime_error("Invalid object id: " + std::to_string(id));
+        }
+    }
+
+private:
+    //----------------------------------------------------------------------------
+    // Private API
+    //----------------------------------------------------------------------------
     bool connect(unsigned int port)
```

Comment:
Why did you make this private? It makes the default constructor somewhat useless by meaning that you can't just create it and only connect if some runtime parameter is set (which I feel I do in the snapshot_bot). I get the horrible feeling that _I_ suggested it - if so it was dumb and I apologise :blush:

Reply:
Oops! I was the one being dumb. I forgot that ``connect`` wouldn't be called in the default constructor.